lib/core.ps1

```diff
@@ -474,116 +474,131 @@ function is_local($path) {
+    }
+    if ($ArgumentList) {
+        if ($FilePath -match '^((cmd|cscript|wscript|msiexec)(\.exe)?|.*\.(bat|cmd|js|vbs|wsf))$') {
+            $Process.StartInfo.Arguments = $ArgumentList -join ' '
+        } elseif ($Process.StartInfo.ArgumentList.Add) {
+            # ArgumentList is supported in PowerShell 6.1 and later (built on .NET Core 2.1+)
+            # ref-1: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=net-6.0
+            # ref-2: https://docs.microsoft.com/en-us/powershell/scripting/whats-new/differences-from-windows-powershell?view=powershell-7.2#net-framework-vs-net-core
+            $ArgumentList | ForEach-Object { $Process.StartInfo.ArgumentList.Add($_) }
+        } else {
+            # escape arguments manually in lower versions, refer to https://docs.microsoft.com/en-us/previous-versions/17w5ykft(v=vs.85)
+            $escapedArgs = $ArgumentList | ForEach-Object {
+                # escape N consecutive backslash(es), which are followed by a double quote, to 2N consecutive ones
+                $s = $_ -replace '(\\+)"', '$1$1"'
+                # escape N consecutive backslash(es), which are at the end of the string, to 2N consecutive ones
+                $s = $s -replace '(\\+)$', '$1$1'
+                # escape double quotes
+                $s = $s -replace '"', '\"'
+                # quote the argument
+                "`"$s`""
+            }
+            $Process.StartInfo.Arguments = $escapedArgs -join ' '
         }
-        $Process.StartInfo.Arguments = $escapedArgs -join ' '
     }
     try {
         [void]$Process.Start()
     } catch {
-        if ($Activity) {
-            Write-Host "error." -ForegroundColor DarkRed
+        if ($Prompt) {
+            Write-Host 'error.' -ForegroundColor DarkRed
         }
         error $_.Exception.Message
         return $false
     }
-    if ($redirectToLogFile) {
-        # we do this to remove a deadlock potential
-        # ref: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?view=netframework-4.5#remarks
-        $stdoutTask = $Process.StandardOutput.ReadToEndAsync()
-        $stderrTask = $Process.StandardError.ReadToEndAsync()
-    }
+    # we do this to remove a deadlock potential
+    # ref: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?view=netframework-4.5#remarks
+    $stdoutTask = $Process.StandardOutput.ReadToEndAsync()
+    $stderrTask = $Process.StandardError.ReadToEndAsync()
     $Process.WaitForExit()
     if ($redirectToLogFile) {
-        Out-UTF8File -FilePath $LogPath -Append -InputObject $stdoutTask.Result
-        Out-UTF8File -FilePath $LogPath -Append -InputObject $stderrTask.Result
+        Out-UTF8File -FilePath $LogName -Append -InputObject $stdoutTask.Result
+        Out-UTF8File -FilePath $LogName -Append -InputObject $stderrTask.Result
+    } else {
+        $stdoutTask.Result | Out-Default
```

Comment:
```suggestion
        if ($stdoutTask.Result) {
            $stdoutTask.Result.Trim() | Out-Default
        }
```

To prevent the output of a random newline on the console